app/src/pages/learn.tsx

```diff
@@ -0,0 +1,184 @@
+
+export default function Learn() {
+  const { data } = useLearningResources();
+  const { tagCategories: tagFilterOptions, typeCategories: typeFilterOptions } =
+    useLearningCategories();
+  const totalResourcesCount = useLearningResourcesTotalCount();
+
+  const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+  const [selectedTypes, setSelectedTypes] = useState<string[]>([]);
+  const [searchText, setSearchText] = useState("");
+
+  const filterData = useCallback(
+    (selectedCategories: string[], selectedTypes: string[], search: string) =>
+      data
+        .filter((resourceItem: LearningCategory) => {
+          const selectedCategoriesEmpty = selectedCategories.length === 0;
+          const selectedTypesEmpty = selectedTypes.length === 0;
+
+          const resourceTags: string[] = resourceItem.items.reduce(
+            (uniqueTags: string[], resource: LearningResource) => {
+              resource.tags.forEach((tag: string) => {
+                if (!uniqueTags.includes(tag)) {
+                  uniqueTags.push(tag);
+                }
+              });
+
+              return uniqueTags;
+            },
+            [] as string[]
+          );
+
+          if (
+            (selectedTypesEmpty || selectedTypes.includes(resourceItem.id)) &&
+            (selectedCategoriesEmpty ||
+              selectedCategories.some((cat) => resourceTags.includes(cat))) &&
+            (search === "" ||
+              resourceItem.items.some(
+                (resource: LearningResource) =>
+                  resource.title.toLowerCase().includes(search.toLowerCase()) ||
+                  resource.description
+                    .toLowerCase()
+                    .includes(search.toLowerCase())
+              ))
+          ) {
+            return true;
+          }
+          return false;
+        })
+        .map(
+          (
+            resourceItem: LearningCategory
+          ): {
+            title: string;
+            id: string;
+            items: LearningResource[];
+          } => ({
+            title: resourceItem.title,
+            id: resourceItem.id,
+            items: resourceItem.items.filter(
+              (resource: LearningResource) =>
+                (selectedCategories.length === 0 ||
+                  resource.tags.some((tag) =>
+                    selectedCategories.includes(tag)
+                  )) &&
+                (selectedTypes.length === 0 ||
+                  selectedTypes.includes(resourceItem.id)) &&
+                (search === "" ||
+                  resource.title.toLowerCase().includes(search.toLowerCase()) ||
+                  resource.description
+                    .toLowerCase()
+                    .includes(search.toLowerCase()))
+            ),
+          })
+        ),
+    [data]
+  );
+
+  return (
+    <div className="flex w-full max-w-[min(calc(100%-1rem),1536px)] flex-col gap-8  2xl:mx-auto">
```

Comment:
This is a minor style change but just to fix the width and remove a space from the `className` string
```html
<div className="flex w-full max-w-[max(calc(100%-1rem),1536px)] flex-col gap-8 2xl:mx-auto">
```